src/SharpCompress/Common/ReaderExtractionEventArgs.cs

```diff
@@ -4,11 +4,13 @@ namespace SharpCompress.Common
 {
     public class ReaderExtractionEventArgs<T> : EventArgs
     {
-        internal ReaderExtractionEventArgs(T entry)
+        internal ReaderExtractionEventArgs(T entry, params object[] paramList)
         {
             Item = entry;
+            ParamList = paramList;
         }
 
         public T Item { get; private set; }
+        public object[] ParamList { get; private set; }
```

Comment:
The idea here was to generalize what could be passed, but am I correct in assuming that you want ReaderProgress here instead of object[] ?